packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.validationRules.tests.js

```diff
@@ -475,6 +475,62 @@ QUnit.test('Changing an validationRules options of an any item does not invalida
     assert.strictEqual(renderComponentSpy.callCount, 0, 'renderComponentSpy.callCount');
 });
 
+[false, true].forEach(hasValidationGroup => {
+    QUnit.test(`form ${hasValidationGroup ? 'with' : 'without'} validation group should validate without errors (T1233487)`, function(assert) {
+        const form = $('#form').dxForm({
+            validationGroup: hasValidationGroup ? 'test' : undefined,
+            formData: {
+                firstName: 'Kyle',
+            },
+            items: [{
+                itemType: 'group',
+                items: [{
+                    dataField: 'firstName',
+                    validationRules: [{ type: 'required' }]
+                }],
+            }],
+        }).dxForm('instance');
+
+        form.option('items[0].items', []);
+
+        try {
+            form.validate();
+            assert.ok(true);
+        } catch(e) {
+            assert.ok(false, 'error is thrown');
+        }
+    });
+});
+
+QUnit.test('validation group should not be removed after its validators are removed (T1233487)', function(assert) {
```

Comment:
[Excess code]
I didn't get what u'r testing here. "validation group should not be removed after its validators are removed" - that's checked in the test above, and you even don't assert it here.
`assert.equal(validator.option('validationGroup'), 'Test', 'validation group should not be removed');` - u'r just checking that option didn't change. But it had no chance to be changed as it's just a group name passed on init ..

Explain please the idea what you tried to test, maybe i'm missing smth

Reply:
Sorry, I thought it was supposed to be a separate test based on what you said above. 

"Also I see that there is no test which checks that custom [ValidationGroup](https://js.devexpress.com/jQuery/Documentation/ApiReference/UI_Components/dxValidationGroup/) is not removed after its validators are removed however this case [is handled](https://github.com/DevExpress/DevExtreme/blob/24_1/packages/devextreme/js/ui/validation_engine.js#L766) in the code. Please add a test for this."

Reply:
Got it. But this scenario is not related to form, only to ValidationGroup component which you even don't use here in your test. Test i said about should check be red if we change this row: https://github.com/DevExpress/DevExtreme/pull/27435#discussion_r1640008793

Reply:
[f8e66aa](https://github.com/DevExpress/DevExtreme/pull/27435/commits/f8e66aaf1e46ace7d774494d1b930d32c43b8b69)

[c874986](https://github.com/DevExpress/DevExtreme/pull/27435/commits/c8749860b9d3db20aa0fb5abd1acd1063392b8c1)